async-query-core/src/main/java/org/opensearch/sql/spark/asyncquery/model/QueryState.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.sql.spark.asyncquery.model;
+
+import java.util.Arrays;
+import java.util.Locale;
+import java.util.Map;
+import java.util.stream.Collectors;
+import lombok.Getter;
+
+@Getter
+public enum QueryState {
```

Comment:
How did we come up with these? Do these also cover emr job status. For eg: `scheduled` when emr is yet to pick up the job.

Reply:
It is coming from StatementState. I think `scheduled` you mentioned is equivalent to `waiting`. Is that more like EMR job status?

Reply:
yeah.